[SofaGUI] Fix Cmake files for out-of-tree compilation

fredroy opened the pull request:
Fix mess in *Config.cmake.in and some dependencies from modularization



______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

fredroy committed:
Fix deps and cmake.in for out of tree compilations

fredroy committed:
Merge branch 'master' of https://github.com/sofa-framework/sofa into fix_outoftree_compil

guparan committed:
[CMake] CLEAN SofaMisc*Config.cmake.in

guparan committed:
[CMake] CLEAN SofaAdvanced*Config.cmake.in

guparan committed:
[CMake] CLEAN SofaGeneralConfig.cmake.in and SofaGuiConfig.cmake.in

guparan committed:
[CMake] Use QUIET REQUIRED in *Config.cmake.in

guparan committed:
[CMake] FIX check_required_components calls

guparan committed:
[CMake] CLEAN variable naming for pluginization

guparan commented:
Here are binaries of this PR to test:
[Linux binaries](https://ci.inria.fr/sofa-ci-dev/job/sofa-custom/547/CI_CONFIG=ubuntu_gcc-7,CI_TYPE=release/artifact/build/SOFA_v20.06.99_Linux.zip)
[Windows binaries](https://ci.inria.fr/sofa-ci-dev/job/sofa-custom/547/CI_CONFIG=windows10_vs-2017,CI_TYPE=release/artifact/build/SOFA_v20.06.99_Win64.zip)

guparan committed:
[CMake] Uniform version for modules

Following Sofa_VERSION

guparan committed:
[SofaMacros] CLEAN version handling for targets

Target version will be:
  TargetName_VERSION if any (eq PROJECT_VERSION)
  - OR -
  PACKAGE_VERSION if any
  - OR -
  Sofa_VERSION if any

guparan committed:
[SofaMacros] FIX missing PROJECT_VERSION

jnbrunet commented:
fast-merge tag + CI ok, I merge 👍
